python/tank_vendor/shotgun_authentication/user.py

```diff
@@ -66,6 +66,16 @@ def create_sg_connection(self):
         """
         return self._impl.create_sg_connection()
 
+    def refresh_credentials(self):
+        """
+        Refreshes the credentials of this user so that they don't expire.
+        If they are expired, you will be prompted for the user's password.
+        """
+        # Make a very simple authenticated request that returns as little information as possible.
+        # If the session token was expired, the ShotgunWrapper returned by create_sg_connection
+        # will take care of the session renewal.
+        self.create_sg_connection().find_one("HumanUser", [])
```

Comment:
very minor: We are using HumanUser to "ping" the server in a couple of places (also in https://github.com/shotgunsoftware/tk-core/pull/157 ) - maybe it could be its own method? `ping_shotgun()` or something?